src/NUnitFramework/framework/Constraints/DictionaryContainsKeyConstraint.cs

```diff
@@ -65,32 +71,138 @@ public override string Description
 
+        /// <summary>
+        /// Flag the constraint to ignore case and return self.
+        /// </summary>
+        [Obsolete("Deprecated, use Does.ContainKey")]
+        public new CollectionItemsEqualConstraint IgnoreCase
+        {
+            get
+            {
+                _deprecatedFlag = true;
+                return base.IgnoreCase;
+            }
+        }
+
         /// <summary>
         /// Test whether the expected key is contained in the dictionary
         /// </summary>
         protected override bool Matches(IEnumerable actual)
         {
-            var dictionary = ConstraintUtils.RequireActual<IDictionary>(actual, nameof(actual));
+            if (_deprecatedFlag)
+            {
+                var dictionary = ConstraintUtils.RequireActual<IDictionary>(actual, nameof(actual));
+                foreach (object obj in dictionary.Keys)
+                    if (ItemsEqual(obj, Expected))
+                        return true;
+
+                return false;
+            }
 
-            foreach (object obj in dictionary.Keys)
-                if (ItemsEqual(obj, Expected))
-                    return true;
+            var method = GetContainsKeyMethod(actual);
+            if (method != null)
+                return (bool)method.Invoke(actual, new[] { Expected });
 
-            return false;
+            throw new ArgumentException("Not a collection supporting ContainsKey method.");
         }
 
         /// <summary>
         /// Flag the constraint to use the supplied predicate function
         /// </summary>
         /// <param name="comparison">The comparison function to use.</param>
         /// <returns>Self.</returns>
+        [Obsolete("Deprecated, use Does.ContainKey")]
         public DictionaryContainsKeyConstraint Using<TCollectionType, TMemberType>(Func<TCollectionType, TMemberType, bool> comparison)
         {
             // reverse the order of the arguments to match expectations of PredicateEqualityComparer
             Func<TMemberType, TCollectionType, bool> invertedComparison = (actual, expected) => comparison.Invoke(expected, actual);
 
+            _deprecatedFlag = true;
             base.Using(EqualityAdapter.For(invertedComparison));
             return this;
         }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied Comparison object.
+        /// </summary>
+        /// <param name="comparer">The IComparer object to use.</param>
+        /// <returns>Self.</returns>
+        [Obsolete("Deprecated, use Does.ContainKey")]
+        public new CollectionItemsEqualConstraint Using<T>(Comparison<T> comparer)
+        {
+            _deprecatedFlag = true;
+            return base.Using(comparer);
+        }
+
+        /// <summary>
+        /// Checks if the key is contained in a "keyed item container".
+        /// </summary>
+        /// <param name="actual">Keyed container.</param>
+        /// <returns>method to call.</returns>
```

Comment:
We [recommend](https://github.com/nunit/docs/wiki/Best-practices-for-XML-documentation#returns-and-value) that you delete the IDE's auto-inserted `<return>` elements because they don't show up in intellisense and we don't do anything else with the XML docs. (If we did, I'd ask for the word 'The' to precede it.)

Also, I think the best option is to rename the parameter `actual` to `keyedItemContainer` and [delete the `<param>`](https://github.com/nunit/docs/wiki/Best-practices-for-XML-documentation#param-and-typeparam).